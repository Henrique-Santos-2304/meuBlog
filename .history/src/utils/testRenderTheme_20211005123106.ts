import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
