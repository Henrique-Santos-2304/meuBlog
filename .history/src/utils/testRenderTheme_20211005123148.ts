import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
