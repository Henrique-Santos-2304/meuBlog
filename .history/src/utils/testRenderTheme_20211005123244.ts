import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
