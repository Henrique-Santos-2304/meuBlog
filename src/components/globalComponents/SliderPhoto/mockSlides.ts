export const mockSlides = {
  images: [
    {
      ariaLabel: "Linguagens,Frameworks ou bibliotecas dominadas",
      title: "Titulo da imagem",
      image: {
        url: "https://res.cloudinary.com/defnibbpl/image/upload/v1635695141/a61237_orkfde_816831fe5f.png",
        alternativeText: "logo da foto1",
      },
      urlProject: "http://projeto1.com",
      ariaLabelButton: "link para codigo",
      buttonText: "ver codigo",
    },
    {
      ariaLabel: "Linguagens,Frameworks ou bibliotecas dominadas",
      title: "titulo da foto",
      image: {
        url: "https://res.cloudinary.com/defnibbpl/image/upload/v1635695141/a61237_orkfde_816831fe5f.png",
        alternativeText: "logo da foto2",
      },
      urlProject: "http://projeto2.com",
      ariaLabelButton: "link para codigo",
      buttonText: "ver codigo",
    },
  ],
  slidesToShow: 1,
  slidesToScroll: 1,
  autoplay: false,
  marginBox: "1rem",
};
