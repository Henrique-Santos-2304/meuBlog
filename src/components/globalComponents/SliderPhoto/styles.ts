import styled from "styled-components";
import Slider from "react-slick";

export const SliderBox = styled(Slider)`
  position: relative;
`;
export const Dots = styled.div`
  bottom: -2.4rem;
  ul {
    &:last-child {
      color: ${({ theme }) => theme.colors.effectsPrimary};
    }
    li {
      color: ${({ theme }) => theme.colors.effectsPrimary};
      &.slick-active {
        button {
          &::before {
            color: ${({ theme }) => theme.colors.effectsPrimary};
          }
        }
      }
      button {
        &::before {
          font-size: 2.3rem;
        }
      }
    }
  }
`;
