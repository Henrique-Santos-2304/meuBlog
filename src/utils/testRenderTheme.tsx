import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
