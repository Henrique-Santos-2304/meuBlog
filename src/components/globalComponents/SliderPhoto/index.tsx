/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-non-null-assertion */
import CardBox from "components/SectionProject/CardBox";
import BoxSkilss from "components/SectionSkilss/BoxSkilss";
import * as S from "./styles";
import { slideImageprops } from "types/typesComponents/typesSlide";

const SliderPhoto = ({
  images,
  receptorCards,
  slidesToShow,
  marginBox,
  responsive,
}: slideImageprops) => {
  const settings = {
    dots: true,
    infinite: false,
    speed: 1000,
    slidesToShow: slidesToShow,
    slidesToScroll: 1,
    arrows: false,
    responsive,
    appendDots: (dots: any) => (
      <S.Dots>
        <ul aria-label="BotÃµes para trocar as fotos"> {dots} </ul>
      </S.Dots>
    ),
  };
  return (
    <S.SliderBox {...settings} aria-label="Slides de fotos">
      {images.map((image) =>
        receptorCards === "skilss" ? (
          <BoxSkilss
            key={image.title}
            title={image.title}
            img={image.img}
            marginBox={marginBox}
            aria-label="Slide de Habilidades"
          />
        ) : (
          <CardBox
            key={image.title}
            title={image.title}
            img={image.img}
            buttonCode={image.buttonCode!}
            marginBox={marginBox}
            aria-label="Slide de Projetos"
          />
        )
      )}
    </S.SliderBox>
  );
};

export default SliderPhoto;
