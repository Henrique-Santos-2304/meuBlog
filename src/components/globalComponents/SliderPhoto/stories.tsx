import { Story, Meta } from "@storybook/react";
import SliderPhoto from ".";

export default {
  title: "SliderPhoto",
  component: SliderPhoto,
} as Meta;

export const Basic: Story = (args) => <SliderPhoto {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <SliderPhoto {...args} />;
