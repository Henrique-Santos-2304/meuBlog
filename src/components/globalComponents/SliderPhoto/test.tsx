import "match-media-mock";
import SliderPhoto from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

const mock = {
  images: [
    {
      img: "https://res.cloudinary.com/defnibbpl/image/upload/v1634679131/storybook_q1s5cl_dafe82d169.svg",
      title: "A primeira imagem do teste",
      buttonCode: "http:buttons.com",
    },
  ],
  marginBox: "1rem",
};
describe("<SliderPhoto />", () => {
  it("should have cards slides ", () => {
    const { container } = renderTheme(
      <SliderPhoto {...mock} slidesToShow={2} receptorCards="skilss" />
    );
    expect(container).toMatchSnapshot();
  });
  it("should have cards slides of the Skills ", () => {
    renderTheme(
      <SliderPhoto {...mock} slidesToShow={2} receptorCards="skilss" />
    );

    const cardSkills = screen.getByLabelText(
      /Linguagens,Frameworks ou bibliotecas dominadas/i
    );
    expect(cardSkills).toBeInTheDocument();
  });
  it("should have cards slides of the Projects ", () => {
    renderTheme(
      <SliderPhoto {...mock} slidesToShow={2} receptorCards="projects" />
    );

    const cardsProjects = screen.getByLabelText(/Titulo do projeto/i);
    expect(cardsProjects).toBeInTheDocument();
  });
});
