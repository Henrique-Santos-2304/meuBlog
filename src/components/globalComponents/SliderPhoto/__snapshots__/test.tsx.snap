// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SliderPhoto /> should have cards slides  1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.55);
  margin: 1rem;
  padding: 1rem 0.5rem;
  border-radius: 1rem;
  position: relative;
}

.c1::before {
  content: "";
  position: absolute;
  width: 0.3rem;
  height: 70%;
  top: 50%;
  right: -1.5rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background: #b7f600;
}

.c2 {
  width: 5rem;
  height: 4rem;
  position: relative;
}

.c3 {
  margin-top: 1.3rem;
  font-size: 1.4rem;
  font-weight: 300;
  text-align: center;
  color: #f8f8f8;
}

.c0 {
  position: relative;
}

@media (min-width:620px) {
  .c0 {
    border-radius: 2rem;
  }
}

<div>
  <div
    class="slick-slider c0 slick-initialized"
  >
    
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              aria-label="Linguagens,Frameworks ou bibliotecas dominadas"
              class="c1"
            >
              <div
                class="c2"
              >
                <div
                  style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                >
                  <img
                    class=""
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </div>
              </div>
              <h3
                class="c3"
              >
                A primeira imagem do teste
              </h3>
            </div>
          </div>
        </div>
      </div>
    </div>
    
    
  </div>
</div>
`;
